packages/react/src/cerbos-provider.tsx

```diff
@@ -0,0 +1,23 @@
+import type { Client, ClientWithPrincipal } from "@cerbos/core";
+// eslint-disable-next-line import/no-extraneous-dependencies -- TODO: not sure about this rule, maybe importing from peer deps should be allowed?
+import type { ReactElement, ReactNode } from "react";
+// eslint-disable-next-line import/no-extraneous-dependencies -- TODO: not sure about this rule, maybe importing from peer deps should be allowed?
+import { createContext } from "react";
+
+export const CerbosContext = createContext<
+  ClientWithPrincipal<Client> | undefined
+>(undefined);
+
+export interface CerbosProviderProps {
+  children: ReactNode;
+  cerbos: ClientWithPrincipal<Client>;
```

Comment:
I think it would be a bit more ergonomic if the props were

```typescript
client: Client;
principal: Principal;
```

and then we did `value={client.withPrincipal(principal)}`